vars/pipelineApprobation.groovy

```diff
@@ -0,0 +1,132 @@
+/* groovylint-disable DuplicateStringLiteral */
+/* groovylint-disable DuplicateNumberLiteral */
+/* groovylint-disable MethodSize */
+/* groovylint-disable NestedBlockDepth */
+import org.jenkinsci.plugins.workflow.steps.FlowInterruptedException
+import org.jenkinsci.plugins.pipeline.modeldefinition.Utils
+
+void call() {
+
+    properties([
+        copyArtifactPermission('*'),
+        parameters([
+            string(
+                name: 'VEGA_CORE_BRANCH', defaultValue: pipelineDefaults.appr.vegaCoreBranch,
+                description: 'Git branch, tag or hash of the vegaprotocol/vega repository'),
+            string(
+                name: 'SPECS_INTERNAL_BRANCH', defaultValue: pipelineDefaults.appr.specsInternalBranch,
+                description: 'Git branch, tag or hash of the vegaprotocol/specs-internal repository'),
+            string(
+                name: 'MULTISIG_CONTROL_BRANCH', defaultValue: pipelineDefaults.appr.multisigControlBranch,
+                description: 'Git branch, tag or hash of the vegaprotocol/MultisigControl repository'),
+            string(
+                name: 'SYSTEM_TESTS_BRANCH', defaultValue: pipelineDefaults.appr.systemTestsBranch,
+                description: 'Git branch, tag or hash of the vegaprotocol/system-tests repository'),
+
+            string(
+                name: 'SPECS_ARG', defaultValue: pipelineDefaults.appr.specsArg,
+                description: '--specs argument value'),
+            string(
+                name: 'TESTS_ARG', defaultValue: pipelineDefaults.appr.testsArg,
+                description: '--tests argument value'),
+            string(
+                name: 'IGNORE_ARG', defaultValue: pipelineDefaults.appr.ignoreArg,
+                description: '--ignore argument value'),
+            string(
+                name: 'OTHER_ARG', defaultValue: pipelineDefaults.appr.otherArg,
+                description: 'Other arguments'),
+
+            string(
+                name: 'APPROBATION_VERSION', defaultValue: pipelineDefaults.appr.approbationVersion,
+                description: 'Released version of Approbation. latest can be used'),
+        ])
+    ])
+
+    echo "params=${params}"
+
+    node {
+        skipDefaultCheckout()
+        cleanWs()
+
+        timestamps {
+            try {
+                timeout(time: 20, unit: 'MINUTES') {
+                    stage('CI config') {
+                        // Printout all configuration variables
+                        sh 'printenv'
+                        echo "params=${params.inspect()}"
+                    }
+                    //
+                    // GIT clone
+                    //
+                    stage('Git Clone') {
+                        parallel([
+                            'vega core': {
+                                dir('vega') {
+                                    gitClone('vega', params.VEGA_CORE_BRANCH)
```

Comment:
In the pipeline output, this looks like:

```
[](https://jenkins.ops.vega.xyz/blue/organizations/jenkins/system-tests%2Fapprobation/detail/approbation/3/pipeline/72#step-73-log-5)[](https://jenkins.ops.vega.xyz/blue/organizations/jenkins/system-tests%2Fapprobation/detail/approbation/3/pipeline/72#step-73-log-6)[](https://jenkins.ops.vega.xyz/blue/organizations/jenkins/system-tests%2Fapprobation/detail/approbation/3/pipeline/72#step-73-log-7)[](https://jenkins.ops.vega.xyz/blue/organizations/jenkins/system-tests%2Fapprobation/detail/approbation/3/pipeline/72#step-73-log-8)[](https://jenkins.ops.vega.xyz/blue/organizations/jenkins/system-tests%2Fapprobation/detail/approbation/3/pipeline/72#step-73-log-9)[](https://jenkins.ops.vega.xyz/blue/organizations/jenkins/system-tests%2Fapprobation/detail/approbation/3/pipeline/72#step-73-log-10)[](https://jenkins.ops.vega.xyz/blue/organizations/jenkins/system-tests%2Fapprobation/detail/approbation/3/pipeline/72#step-73-log-11)[](https://jenkins.ops.vega.xyz/blue/organizations/jenkins/system-tests%2Fapprobation/detail/approbation/3/pipeline/72#step-73-log-12)[2022-03-17T05:32:36.452Z] Project branches
[2022-03-17T05:32:36.452Z]   MultisigControl: Detached: 4989720c08ae91247b7d672f9f3214aeee3135d7 @ 4989720
[2022-03-17T05:32:36.452Z]   Could not check git branch for MultisigControl@tmp
[2022-03-17T05:32:36.452Z]   specs-internal: Detached: 1b2ca48513a6d8f99338b6acca278daff1268b90 @ 1b2ca48
[2022-03-17T05:32:36.452Z]   Could not check git branch for specs-internal@tmp
[2022-03-17T05:32:36.452Z]   system-tests: Detached: ff66754d0d026eebb54f0a1eb53aa9c03c2d35f3 @ ff66754
[2022-03-17T05:32:36.452Z]   Could not check git branch for system-tests@tmp
[2022-03-17T05:32:36.452Z]   vega: Detached: 92e38aabeddb94c249cf76109960c75f884e37ed @ 92e38aa
[2022-03-17T05:32:36.452Z]   Could not check git branch for vega@tmp
```

This text output is produced here: https://github.com/vegaprotocol/approbation/blob/6cb1fae231737bd7507ba352a61106c31da8bb00/src/lib/get-project-branches.js#L27-L48

What I don't understand is why this produces commit hashes and not meaningful branch names - do you have any idea? 

Again, this isn't a blocker. The branch *name* becomes less interesting when it's running in CI, I added that feature so that while I was running locally, it would be clear in output I posted in slack what branches of projects I was running because `system-tests` & `multisigcontrol` both had more tests in non-default branches. Given CI specifies the branches, and both repos have since had their tests merged, it's not a huge deal. But if it's an easy fix, let's fix it.

Reply:
I think it produces hashes because Jenkins is using git hashed instead of branches from a practical point of view :(

It switches to a particular commit from the branch (sent in webhook from Github or collected by repository crawler)

Let's say you have the following scenarios:
- You committed quickly 3 times, Jenkins runs 3 builds, If you have a commit hash it's clear what changes are building, if you use branch name it is not so obvious :) 

- You override some changes with the `push --force`, and if you use branch name you have no chance to figure out what happened :) 

I think this is possible to use branch names but it's kind of a trick to call `git checkout` manually :'( 

But it will slow a little down your job because if you are fetching only commit hash you do not need to know about other branches and commit outside of your branch -> This is another reason why Jenkins is using hash because it collects what is required only :(